pkg/webui/components/qr-model-button/index.js

```diff
@@ -0,0 +1,124 @@
+// Copyright © 2021 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useCallback } from 'react'
+import { defineMessages } from 'react-intl'
+
+import Link from '@ttn-lw/components/link'
+import ModalButton from '@ttn-lw/components/button/modal-button'
+
+import Message from '@ttn-lw/lib/components/message'
+import ErrorMessage from '@ttn-lw/lib/components/error-message'
+
+import PropTypes from '@ttn-lw/lib/prop-types'
+import sharedMessages from '@ttn-lw/lib/shared-messages'
+
+import DataSheet from '../data-sheet'
+import QR from '../qr'
+
+import style from './qr-modal-button.styl'
+
+const QrScanDoc = (
+  <Link.Anchor external secondary href="https://www.thethingsindustries.com/docs/">
+    Having trouble?
+  </Link.Anchor>
+)
+
+const m = defineMessages({
+  scanEndDevice: 'Scan end device QR code',
+  scanEndDeviceContinue: 'Please scan the QR code to continue. {qrScanDoc}',
+  invalidData: 'Invalid QR data please try again',
+  confirm: 'Confirm',
+  scanAgain: 'Scan Again',
+})
+
+const QRModalButton = props => {
+  const { message, onApprove, onCancel, onRead, approveData } = props
+
+  const handleRead = useCallback(
+    val => {
+      onRead(val)
+    },
+    [onRead],
+  )
+
+  const modelData = () => (
+    <div className={style.qrmodel}>
+      {approveData ? (
+        approveData.valid ? (
+          <DataSheet data={approveData.data} />
+        ) : (
+          <ErrorMessage content={m.invalidData} />
+        )
+      ) : (
+        <>
+          <QR onChange={handleRead} />
+          <Message
+            content={m.scanEndDeviceContinue}
+            values={{ qrScanDoc: QrScanDoc }}
+            component="span"
+          />
+        </>
+      )}
+    </div>
+  )
+
+  const cancelButtonProps = () => {
+    const props = {}
+
+    if (approveData) {
+      props.onClick = onCancel
+    }
+
+    return props
+  }
+
+  return (
+    <ModalButton
+      type="button"
+      icon="camera_alt"
+      onCancel={onCancel}
+      onApprove={onApprove}
+      message={message}
+      modalData={{
+        title: m.scanEndDevice,
+        children: modelData(),
+        buttonMessage: m.confirm,
+        approveButtonProps: {
+          disabled: !approveData?.valid,
+        },
+        cancelButtonMessage: approveData ? m.scanAgain : sharedMessages.cancel,
+        cancelButtonProps: cancelButtonProps(),
```

Comment:
```suggestion
        cancelButtonProps: {
          disabled: Boolean(approveData),
        },
```

Reply:
@kschiffer Do you still want the cancel button to return the user to the video input if we are on the approval data screen?